src/main/java/seedu/address/ui/PersonListPanel.java

```diff
@@ -50,10 +63,28 @@ private void setEventHandlerForSelectionChangeEvent() {
                     if (newValue != null) {
                         logger.fine("Selection in person list panel changed to : '" + newValue + "'");
                         raise(new PersonPanelSelectionChangedEvent(newValue));
+                        raise(new NewResultAvailableEvent(getSelectionMessage(personListView.getSelectionModel()
+                                .getSelectedIndex() + 1), false));
                     }
                 });
     }
 
+    private String getSelectionMessage(int oneBasedIndex) {
+        switch (currentListName) {
+        case BlacklistCommand.COMMAND_WORD:
+            return BLACKLIST_NAME_DISPLAY_FORMAT + String.format(MESSAGE_SELECT_PERSON_SUCCESS, oneBasedIndex);
+
+        case WhitelistCommand.COMMAND_WORD:
+            return WHITELIST_NAME_DISPLAY_FORMAT + String.format(MESSAGE_SELECT_PERSON_SUCCESS, oneBasedIndex);
+
+        case OverdueListCommand.COMMAND_WORD:
+            return OVERDUELIST_NAME_DISPLAY_FORMAT + String.format(MESSAGE_SELECT_PERSON_SUCCESS, oneBasedIndex);
+
+        default:
+            return MASTERLIST_NAME_DISPLAY_FORMAT + String.format(MESSAGE_SELECT_PERSON_SUCCESS, oneBasedIndex);
+        }
+    }
+
```

Comment:
In my previous PR(the one that had ListObserver), you mentioned we should bring ListObserver outside of Command Class. I will see what I can do about it because the changes you are doing here is supposed to be done by the ListObserver.